src/connectors/__test__/exchangeRate.test.ts

```diff
@@ -0,0 +1,126 @@
+import { CacheService, ExchangeRate } from 'connectors'
+
+// stub data
+
+const coingeckoAPIData = {
+  likecoin: {
+    hkd: 0.01919234,
+    twd: 0.07643,
+    usd: 0.0024524,
+    last_updated_at: 1668738838,
+  },
+  tether: {
+    hkd: 7.82,
+    twd: 31.15,
+    usd: 0.999504,
+    last_updated_at: 1668738623,
+  },
+}
+
+const exchangeRatesDataAPIData = {
+  base: 'HKD',
+  date: '2022-11-18',
+  rates: {
+    HKD: 1,
+    TWD: 3.982979,
+    USD: 0.127826,
+  },
+  success: true,
+  timestamp: 1668752883,
+}
+
+const rates = [
+  {
+    from: 'LIKE',
+    to: 'TWD',
+    rate: 0.07643,
+    updatedAt: new Date('2022-11-18T02:33:58.000Z'),
+  },
+  {
+    from: 'LIKE',
+    to: 'HKD',
+    rate: 0.01919234,
+    updatedAt: new Date('2022-11-18T02:33:58.000Z'),
+  },
+  {
+    from: 'LIKE',
+    to: 'USD',
+    rate: 0.0024524,
+    updatedAt: new Date('2022-11-18T02:33:58.000Z'),
+  },
+  {
+    from: 'USDT',
+    to: 'TWD',
+    rate: 31.15,
+    updatedAt: new Date('2022-11-18T02:30:23.000Z'),
+  },
+  {
+    from: 'USDT',
+    to: 'HKD',
+    rate: 7.82,
+    updatedAt: new Date('2022-11-18T02:30:23.000Z'),
+  },
+  {
+    from: 'USDT',
+    to: 'USD',
+    rate: 0.999504,
+    updatedAt: new Date('2022-11-18T02:30:23.000Z'),
+  },
+  {
+    from: 'HKD',
+    to: 'TWD',
+    rate: 3.982979,
+    updatedAt: new Date('2022-11-18T06:28:03.000Z'),
+  },
+  {
+    from: 'HKD',
+    to: 'HKD',
+    rate: 1,
+    updatedAt: new Date('2022-11-18T06:28:03.000Z'),
+  },
+  {
+    from: 'HKD',
+    to: 'USD',
+    rate: 0.127826,
+    updatedAt: new Date('2022-11-18T06:28:03.000Z'),
+  },
+]
+
+describe('exchangeRate', () => {
+  const exchangeRate = new ExchangeRate()
+  beforeEach(() => {
+    // mock
+    exchangeRate.expire = 3000 // 3 seconds
```

Comment:
Looks like I got it wrong, `expire` is second-basis, so `3000` is 3000 seconds...